ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-items/workspaces-items.tsx

```diff
@@ -112,7 +112,12 @@ const WorkspacesItems = (props: Props) => {
           </ModifiedButton>
           {props.workspaces.map(workspace => {
             return (
-              <WorkspaceItemRoot key={workspace.id}>
+              <WorkspaceItemRoot
+                key={
+                  workspace.id ||
+                  Math.floor(Math.random() * Number.MAX_SAFE_INTEGER)
```

Comment:
You could also use the workspace index which is at least more stable than a random number

Reply:
the workspace index in the collection? I feel like that could cause issues

Reply:
I guess someone caused you to update this, but I think your original solution of filtering out those without an id was clever and correct.